plugin/pkg/scheduler/schedulercache/cache_test.go

```diff
@@ -0,0 +1,467 @@
+		cleanupTime time.Time
+
+		wNodeInfo *NodeInfo
+	}{{ // assumed pod would expires
+		pods:        []*api.Pod{makeBasePod(nodeName, "test", "100m", "500", []api.ContainerPort{{HostPort: 80}})},
+		assumedTime: []time.Time{now},
+		cleanupTime: now.Add(2 * ttl),
+		wNodeInfo:   nil,
+	}, { // first one would expire, second one would not.
+		pods:        []*api.Pod{testPods[0], testPods[1]},
+		assumedTime: []time.Time{now, now.Add(3 * ttl / 2)},
+		cleanupTime: now.Add(2 * ttl),
+		wNodeInfo: &NodeInfo{
+			requestedResource: &Resource{
+				MilliCPU: 200,
+				Memory:   1024,
+			},
+			nonzeroRequest: &Resource{
+				MilliCPU: 200,
+				Memory:   1024,
+			},
+			pods: []*api.Pod{testPods[1]},
+		},
+	}}
+
+	for i, tt := range tests {
+		cache := newSchedulerCache(ttl, time.Second, nil)
+
+		for j, pod := range tt.pods {
+			err := cache.assumePodIfBindSucceed(pod, alwaysTrue, tt.assumedTime[j])
+			if err != nil {
+				t.Fatalf("assumePod failed: %v", err)
+			}
+		}
+		// pods that have assumedTime + ttl < cleanupTime will get expired and removed
+		cache.cleanupAssumedPods(tt.cleanupTime)
+		n := cache.nodes[nodeName]
+		if !reflect.DeepEqual(n, tt.wNodeInfo) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+	}
+}
+
+// TestAddPodWillConfirm tests that a pod being Add()ed will be confirmed if assumed.
+// The pod info should still exist after manually expiring unconfirmed pods.
+func TestAddPodWillConfirm(t *testing.T) {
+	nodeName := "node"
+	ttl := 10 * time.Second
+	basePod := makeBasePod(nodeName, "test", "100m", "500", []api.ContainerPort{{HostPort: 80}})
+	tests := []struct {
+		podToAssume *api.Pod
+		podToAdd    *api.Pod
+
+		wNodeInfo *NodeInfo
+	}{{ // Pod is assumed. It should be confirmed.
+		podToAssume: basePod,
+		podToAdd:    basePod,
+		wNodeInfo: &NodeInfo{
+			requestedResource: &Resource{
+				MilliCPU: 100,
+				Memory:   500,
+			},
+			nonzeroRequest: &Resource{
+				MilliCPU: 100,
+				Memory:   500,
+			},
+			pods: []*api.Pod{basePod},
+		},
+	}}
+
+	now := time.Now()
+	for i, tt := range tests {
+		cache := newSchedulerCache(ttl, time.Second, nil)
+		err := cache.assumePodIfBindSucceed(tt.podToAssume, alwaysTrue, now)
+		if err != nil {
+			t.Fatalf("assumePod failed: %v", err)
+		}
+		err = cache.AddPod(tt.podToAdd)
+		if err != nil {
+			t.Fatalf("AddPod failed: %v", err)
+		}
+		cache.cleanupAssumedPods(now.Add(2 * ttl))
+		// check after expiration. confirmed pods shouldn't be expired.
+		n := cache.nodes[nodeName]
+		if !reflect.DeepEqual(n, tt.wNodeInfo) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+	}
+}
+
+// TestAddPodAfterExpiration tests that a pod being Add()ed will be added back if expired.
+func TestAddPodAfterExpiration(t *testing.T) {
+	nodeName := "node"
+	ttl := 10 * time.Second
+	basePod := makeBasePod(nodeName, "test", "100m", "500", []api.ContainerPort{{HostPort: 80}})
+	tests := []struct {
+		podToAssume *api.Pod
+		podToAdd    *api.Pod
```

Comment:
Please generalize both to be []*api.Pod (to make more sophisticated test possible)